portal-impl/src/com/liferay/portal/service/impl/ResourcePermissionLocalServiceImpl.java

```diff
@@ -362,8 +362,11 @@ public ResourcePermission fetchResourcePermission(
 			resourcePermissions.size());
 
 		for (ResourcePermission resourcePermission : resourcePermissions) {
+			Set<String> emptySet = Collections.emptySet();
+
 			if (resourcePermission.getActionIds() == 0) {
-				continue;
+				roleIdsToActionIds.put(
```

Comment:
When I put Collections.emptySet() inside the call to put, the code won't compile. It gets the following error:

> D:\Repos\liferay-portal\portal-impl\src\com\liferay\portal\service\impl\ResourcePermissionLocalServiceImpl.java:366: error: method put in interface Map\<K,V\> cannot be applied to given types;
>     [javac]                             roleIdsToActionIds.put(
>     [javac]                                               ^
>     [javac]   required: Long,Set\<String\>
>     [javac]   found: long,Set\<Object\>
>     [javac]   reason: actual argument Set\<Object\> cannot be converted to Set\<String\> by method invocation conversion

So I'm only going to fix the first issue.